src/core/etl/src/Flow/ETL/PHP/Type/Types.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\PHP\Type;
+
+use Flow\ETL\Exception\InvalidArgumentException;
+use Flow\ETL\PHP\Type\Native\NullType;
+
+final class Types implements \Countable
+{
+    private readonly array $types;
+
+    private function __construct(Type ...$types)
+    {
+        $this->types = \array_map(
+            fn (string $type) : Type => \unserialize($type),
+            \array_unique(
+                \array_map(
+                    fn (Type $type) : string => \serialize($type),
+                    \array_filter($types, fn (Type $type) : bool => !$type instanceof NullType)
```

Comment:
This should be requested intentionally, not filtered out by default, because we are losing knowledge about nulls in array